src/dds/rs-dds-sensor-proxy.cpp

```diff
@@ -564,6 +581,41 @@ void dds_sensor_proxy::add_option( std::shared_ptr< realdds::dds_option > option
         } );
     register_option( option_id, opt );
     _options_watcher.register_option( option_id, opt );
+
+    if( std::dynamic_pointer_cast< realdds::dds_rect_option >( option ) && option->get_name() == "Region of Interest" )
+    {
+        if( _roi_support )
```

Comment:
Please move into `handle_roi_support` function

Reply:
I just took the class definition out of the function, that made it a 3-line block inside the if.